### PR TITLE
update code example in raft-in-rust

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### CODE REVIEW

File: implement-raft-in-rust.md [hunk 1]
```diff
@@ -97,57 +97,60 @@ Other important fields like `check_quorum` and `pre_vote` are used to avoid the
...
+        handle_raft_ready(raft.ready());
+    }
+}
+```
+
+There are three steps to this process before `handle_raft_ready`:
+
+1. You can call the `step` function when you receive the Raft messages from other nodes. 
+
+    Calling `Raft::step` will changes memory state of the `Raft`.
```
Comment (dcalvin) [inline implement-raft-in-rust.md]:
```suggestion
    Calling `Raft::step`  changes the memory state of `Raft`.

```

File: implement-raft-in-rust.md [hunk 2]
```diff
@@ -97,57 +97,60 @@ Other important fields like `check_quorum` and `pre_vote` are used to avoid the
...
 
 2. Use the `propose` function to drive the Raft node when the client sends a request to the Raft server. You can call `propose` to add the request to the Raft log explicitly.
 
     In most cases, the client needs to wait for a response for the request. For example, if the client writes a value to a key and wants to know whether the write succeeds or not, but the write flow is asynchronous in Raft, so the write log entry must be replicated to other followers, then committed and at last applied to the state machine, so here we need a way to notify the client after the write is finished. 
   
     One simple way is to use a unique ID for the client request, and save the associated callback function in a hash map. When the log entry is applied, we can get the ID from the decoded entry, call the corresponding callback, and notify the client. 
 
-3. You can call the `step` function when you receive the Raft messages from other nodes. 
+3. You need a timer to run the Raft node regularly. In the example we use Rust channel `recv_timeout`.
```
Comment (dcalvin) [inline implement-raft-in-rust.md]:
```suggestion
3. You need a timer to run the Raft node regularly. In the above example, we used Rust channel `recv_timeout`.

```

**Comment by dcalvin** [comment]
You can disable the markdown lint for the file by adding `<!-- markdownlint-disable -->`

**Comment by WT-Liu** [comment]
@dcalvin PTAL and thanks!

**Comment by dcalvin** [comment]
Need to fix the md lint error. `kost-stack.md:24:1007 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]`

**Pull request merged**